internal/controller/taggable_controller.go

```diff
@@ -213,15 +219,15 @@ func (r *TaggableResourceReconciler[T, P, PT]) generateBinding(resource PT, proj
 		Spec: tagsv1alpha1.TagsLocationTagBindingSpec{
 			Location: r.MetadataProvider.GetResourceLocation(resource),
 			ParentRef: ccv1alpha1.ResourceRef{
-				External: r.MetadataProvider.GetResourceID(projectID, resource),
+				External: r.MetadataProvider.GetResourceID(projectInfo, resource),
 			},
 			TagValueRef: ccv1alpha1.ResourceRef{
 				External: tagValueID,
 			},
 		},
 	}
-	if projectID != "" {
-		binding.Annotations[projectIDAnnotation] = projectID
+	if projectInfo.ProjectId != "" {
```

Comment:
I guess this project id will never be an empty string, since the API always returns something? In that case we might as well remove this